Python 3.13 support

When running on a macOS runner with the `macos-latest` tag, it currently uses Python 3.13 by default and has the following errors when trying to run `publish-unit-test-result-action`:

```
  Building wheels for collected packages: lxml, cffi
    Building wheel for lxml (pyproject.toml): started
    Building wheel for lxml (pyproject.toml): finished with status 'done'
    Created wheel for lxml: filename=lxml-5.1.0-cp313-cp313-macosx_14_0_arm64.whl size=1595644 sha256=959de9c98b1c8ead73a0c5182182a7ab6d03419877245a8e763b8ae39c09d60d
    Stored in directory: /Users/runner/Library/Caches/pip/wheels/61/ee/c7/8efaa07f329bd1007fb1f352458a424aa2e7358b1ff215f374
    Building wheel for cffi (pyproject.toml): started
    Building wheel for cffi (pyproject.toml): finished with status 'error'
    error: subprocess-exited-with-error
    
    × Building wheel for cffi (pyproject.toml) did not run successfully.
    │ exit code: 1
    ╰─> [48 lines of output]
        running bdist_wheel
        running build
        running build_py
        creating build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/_imp_emulation.py -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/backend_ctypes.py -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/error.py -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/setuptools_ext.py -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/_shimmed_dist_utils.py -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/__init__.py -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/cffi_opcode.py -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/vengine_gen.py -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/pkgconfig.py -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/model.py -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/ffiplatform.py -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/api.py -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/vengine_cpy.py -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/commontypes.py -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/lock.py -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/recompiler.py -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/cparser.py -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/verifier.py -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/_cffi_include.h -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/parse_c_type.h -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/_embedding.h -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        copying src/cffi/_cffi_errors.h -> build/lib.macosx-14.0-arm64-cpython-313/cffi
        running build_ext
        building '_cffi_backend' extension
        creating build/temp.macosx-14.0-arm64-cpython-313/src/c
        clang -fno-strict-overflow -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -DFFI_BUILDING=1 -DUSE__THREAD -DHAVE_SYNC_SYNCHRONIZE -I/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/usr/include/ffi -I/Users/runner/work/DeepSea/DeepSea/enricomi-publish-action-venv/include -I/opt/homebrew/opt/python@3.13/Frameworks/Python.framework/Versions/3.13/include/python3.13 -c src/c/_cffi_backend.c -o build/temp.macosx-14.0-arm64-cpython-313/src/c/_cffi_backend.o -iwithsysroot/usr/include/ffi
        src/c/_cffi_backend.c:4532:22: warning: 'Py_FileSystemDefaultEncoding' is deprecated [-Wdeprecated-declarations]
                             Py_FileSystemDefaultEncoding, &filename_or_null, &flags))
                             ^
        /opt/homebrew/opt/python@3.13/Frameworks/Python.framework/Versions/3.13/include/python3.13/fileobject.h:22:1: note: 'Py_FileSystemDefaultEncoding' has been explicitly marked deprecated here
        Py_DEPRECATED(3.12) PyAPI_DATA(const char *) Py_FileSystemDefaultEncoding;
        ^
        /opt/homebrew/opt/python@3.13/Frameworks/Python.framework/Versions/3.13/include/python3.13/pyport.h:251:54: note: expanded from macro 'Py_DEPRECATED'
        #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                             ^
        src/c/_cffi_backend.c:6121:9: error: call to undeclared function '_PyErr_WriteUnraisableMsg'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
                _PyErr_WriteUnraisableMsg(PyText_AS_UTF8(s), NULL);
                ^
        src/c/_cffi_backend.c:6121:9: note: did you mean 'PyErr_WriteUnraisable'?
        /opt/homebrew/opt/python@3.13/Frameworks/Python.framework/Versions/3.13/include/python3.13/pyerrors.h:233:18: note: 'PyErr_WriteUnraisable' declared here
        PyAPI_FUNC(void) PyErr_WriteUnraisable(PyObject *);
                         ^
        1 warning and 1 error generated.
        error: command '/usr/bin/clang' failed with exit code 1
        [end of output]
    
    note: This error originates from a subprocess, and is likely not a problem with pip.
    ERROR: Failed building wheel for cffi
  Successfully built lxml
  Failed to build cffi
  ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (cffi)
```

This appears to be in the cffi dependency, which added support for Python 3.13 in 1.17.

--- a/.github/actions/test/action.yml
+++ b/.github/actions/test/action.yml
@@ -19,16 +19,16 @@ runs:
         sudo apt-get update
         sudo apt-get install language-pack-en language-pack-de
       shell: bash
-  
+
     - name: Setup Python
       if: inputs.python-version != 'installed'
       uses: actions/setup-python@v5
       with:
         python-version: ${{ inputs.python-version }}
-  
+
     - name: Checkout
       uses: actions/checkout@v4
-  
+
     - name: Detect OS
       id: os
       env:
@@ -47,7 +47,7 @@ runs:
         esac
         echo "date=$(date +%Y%m%d 2> /dev/null || true)" >> $GITHUB_OUTPUT
       shell: bash
-  
+
     - name: Cache PIP Packages
       uses: actions/cache@v4
       id: cache
@@ -59,7 +59,7 @@ runs:
           ${{ inputs.os }}-pip-test-${{ inputs.python-version }}-${{ hashFiles('**/requirements.txt', '**/constraints.txt') }}-
           ${{ inputs.os }}-pip-test-${{ inputs.python-version }}-
           ${{ inputs.os }}-pip-test-
-  
+
     - name: Install Python dependencies
       run: |
         python -m venv "$RUNNER_TEMP/venv"
@@ -88,7 +88,7 @@ runs:
         python -m pip cache list || true
 
         # assert no whl files have been built
-        if [[ "$python_minor_version" != "3.13" ]] && python -m pip cache info && [[ "$(python -m pip cache info | grep "Number of .*wheels:")" != "Number of "*"wheels: 0" ]]
+        if python -m pip cache info && [[ "$(python -m pip cache info | grep "Number of .*wheels:")" != "Number of "*"wheels: 0" ]]
         then
           echo "Dependency whl files have been built"
           exit 1

--- a/.github/actions/test/action.yml
+++ b/.github/actions/test/action.yml
@@ -88,7 +88,7 @@ runs:
         python -m pip cache list || true
 
         # assert no whl files have been built
-        if python -m pip cache info && [[ "$(python -m pip cache info | grep "Number of .*wheels:")" != "Number of "*"wheels: 0" ]]
+        if [[ "$python_minor_version" != "3.13" ]] && python -m pip cache info && [[ "$(python -m pip cache info | grep "Number of .*wheels:")" != "Number of "*"wheels: 0" ]]
         then
           echo "Dependency whl files have been built"
           exit 1

--- a/.github/workflows/ci-cd.yml
+++ b/.github/workflows/ci-cd.yml
@@ -38,7 +38,7 @@ jobs:
     uses: "./.github/workflows/test-os.yml"
     with:
       os: '["macos-12", "macos-13", "macos-14", "macos-15"]'
-      python-version: '["3.10", "3.11", "3.12", "3.13.0-alpha.3", "installed"]'
+      python-version: '["3.10", "3.11", "3.12", "3.13.0-beta.3", "installed"]'
       include: >
         [
           {"os": "macos-12", "python-version": "3.8"},
@@ -52,7 +52,7 @@ jobs:
     uses: "./.github/workflows/test-os.yml"
     with:
       os: '["ubuntu-20.04", "ubuntu-22.04", "ubuntu-24.04"]'
-      python-version: '["3.9", "3.10", "3.11", "3.12", "3.13.0-alpha.3", "installed"]'
+      python-version: '["3.9", "3.10", "3.11", "3.12", "3.13.0-beta.3", "installed"]'
       include: >
         [
           {"os": "ubuntu-20.04", "python-version": "3.7"},
@@ -65,7 +65,7 @@ jobs:
     uses: "./.github/workflows/test-os.yml"
     with:
       os: '["windows-2019", "windows-2022"]'
-      python-version: '["3.8", "3.9", "3.10", "3.11", "3.12", "3.13.0-alpha.3", "installed"]'
+      python-version: '["3.8", "3.9", "3.10", "3.11", "3.12", "3.13.0-beta.3", "installed"]'
       include: '[{"os": "windows-2019", "python-version": "3.7"}]'
 
   publish:

--- a/.github/workflows/ci-cd.yml
+++ b/.github/workflows/ci-cd.yml
@@ -38,7 +38,7 @@ jobs:
     uses: "./.github/workflows/test-os.yml"
     with:
       os: '["macos-12", "macos-13", "macos-14", "macos-15"]'
-      python-version: '["3.10", "3.11", "3.12", "3.13.0-beta.3", "installed"]'
+      python-version: '["3.10", "3.11", "3.12", "3.13", "installed"]'
       include: >
         [
           {"os": "macos-12", "python-version": "3.8"},
@@ -52,7 +52,7 @@ jobs:
     uses: "./.github/workflows/test-os.yml"
     with:
       os: '["ubuntu-20.04", "ubuntu-22.04", "ubuntu-24.04"]'
-      python-version: '["3.9", "3.10", "3.11", "3.12", "3.13.0-beta.3", "installed"]'
+      python-version: '["3.9", "3.10", "3.11", "3.12", "3.13", "installed"]'
       include: >
         [
           {"os": "ubuntu-20.04", "python-version": "3.7"},
@@ -65,7 +65,7 @@ jobs:
     uses: "./.github/workflows/test-os.yml"
     with:
       os: '["windows-2019", "windows-2022"]'
-      python-version: '["3.8", "3.9", "3.10", "3.11", "3.12", "3.13.0-beta.3", "installed"]'
+      python-version: '["3.8", "3.9", "3.10", "3.11", "3.12", "3.13", "installed"]'
       include: '[{"os": "windows-2019", "python-version": "3.7"}]'
 
   publish:

--- a/.github/workflows/ci-cd.yml
+++ b/.github/workflows/ci-cd.yml
@@ -38,7 +38,7 @@ jobs:
     uses: "./.github/workflows/test-os.yml"
     with:
       os: '["macos-12", "macos-13", "macos-14", "macos-15"]'
-      python-version: '["3.10", "3.11", "3.12", "installed"]'
+      python-version: '["3.10", "3.11", "3.12", "3.13.0-alpha.3", "installed"]'
       include: >
         [
           {"os": "macos-12", "python-version": "3.8"},
@@ -52,7 +52,7 @@ jobs:
     uses: "./.github/workflows/test-os.yml"
     with:
       os: '["ubuntu-20.04", "ubuntu-22.04", "ubuntu-24.04"]'
-      python-version: '["3.9", "3.10", "3.11", "3.12", "installed"]'
+      python-version: '["3.9", "3.10", "3.11", "3.12", "3.13.0-alpha.3", "installed"]'
       include: >
         [
           {"os": "ubuntu-20.04", "python-version": "3.7"},
@@ -65,7 +65,7 @@ jobs:
     uses: "./.github/workflows/test-os.yml"
     with:
       os: '["windows-2019", "windows-2022"]'
-      python-version: '["3.8", "3.9", "3.10", "3.11", "3.12", "installed"]'
+      python-version: '["3.8", "3.9", "3.10", "3.11", "3.12", "3.13.0-alpha.3", "installed"]'
       include: '[{"os": "windows-2019", "python-version": "3.7"}]'
 
   publish:

--- a/composite/action.yml
+++ b/composite/action.yml
@@ -218,7 +218,7 @@ runs:
       continue-on-error: true
       with:
         path: ${{ steps.os.outputs.pip-cache }}
-        key: enricomi-publish-action-${{ runner.os }}-${{ runner.arch }}-pip-${{ steps.python.outputs.version }}-e594996205319a7990b3a4ec677d10a3
+        key: enricomi-publish-action-${{ runner.os }}-${{ runner.arch }}-pip-${{ steps.python.outputs.version }}-1a7833c49f5935a970bc661e3c45aa2a
 
     - name: Create virtualenv
       id: venv

--- a/linux/action.yml
+++ b/linux/action.yml
@@ -194,7 +194,7 @@ runs:
       continue-on-error: true
       with:
         path: '~/.cache/pip'
-        key: enricomi-publish-action-${{ runner.os }}-${{ runner.arch }}-pip-${{ steps.python.outputs.version }}-e594996205319a7990b3a4ec677d10a3
+        key: enricomi-publish-action-${{ runner.os }}-${{ runner.arch }}-pip-${{ steps.python.outputs.version }}-1a7833c49f5935a970bc661e3c45aa2a
 
     - name: Create virtualenv
       id: venv

--- a/macos/action.yml
+++ b/macos/action.yml
@@ -194,7 +194,7 @@ runs:
       continue-on-error: true
       with:
         path: '~/Library/Caches/pip'
-        key: enricomi-publish-action-${{ runner.os }}-${{ runner.arch }}-pip-${{ steps.python.outputs.version }}-e594996205319a7990b3a4ec677d10a3
+        key: enricomi-publish-action-${{ runner.os }}-${{ runner.arch }}-pip-${{ steps.python.outputs.version }}-1a7833c49f5935a970bc661e3c45aa2a
 
     - name: Create virtualenv
       id: venv

--- a/python/requirements-3.10.txt
+++ b/python/requirements-3.10.txt
@@ -1,6 +1,6 @@
 humanize==4.9.0
 junitparser==3.1.2
-lxml==5.1.0
+lxml==5.3.0
 psutil==5.9.8
 PyGithub==2.2.0
   Deprecated==1.2.14

--- a/python/requirements-3.10.txt
+++ b/python/requirements-3.10.txt
@@ -7,7 +7,7 @@ PyGithub==2.2.0
     wrapt==1.16.0
   PyJWT==2.8.0
   PyNaCl==1.5.0
-    cffi==1.16.0
+    cffi==1.17.1
       pycparser==2.22
   requests==2.32.3
     certifi==2024.7.4

--- a/python/requirements-3.11.txt
+++ b/python/requirements-3.11.txt
@@ -1,6 +1,6 @@
 humanize==4.9.0
 junitparser==3.1.2
-lxml==5.1.0
+lxml==5.3.0
 psutil==5.9.8
 PyGithub==2.2.0
   Deprecated==1.2.14

--- a/python/requirements-3.11.txt
+++ b/python/requirements-3.11.txt
@@ -7,7 +7,7 @@ PyGithub==2.2.0
     wrapt==1.16.0
   PyJWT==2.8.0
   PyNaCl==1.5.0
-    cffi==1.16.0
+    cffi==1.17.1
       pycparser==2.22
   requests==2.32.3
     certifi==2024.7.4

--- a/python/requirements-3.12.txt
+++ b/python/requirements-3.12.txt
@@ -1,6 +1,6 @@
 humanize==4.9.0
 junitparser==3.1.2
-lxml==5.1.0
+lxml==5.3.0
 psutil==5.9.8
 PyGithub==2.2.0
   Deprecated==1.2.14

--- a/python/requirements-3.12.txt
+++ b/python/requirements-3.12.txt
@@ -7,7 +7,7 @@ PyGithub==2.2.0
     wrapt==1.16.0
   PyJWT==2.8.0
   PyNaCl==1.5.0
-    cffi==1.16.0
+    cffi==1.17.1
       pycparser==2.22
   requests==2.32.3
     certifi==2024.7.4

--- a/python/requirements-3.13.txt
+++ b/python/requirements-3.13.txt
@@ -1,6 +1,6 @@
 humanize==4.9.0
 junitparser==3.1.2
-lxml==5.1.0
+lxml==5.3.0
 psutil==5.9.8
 PyGithub==2.2.0
   Deprecated==1.2.14

--- a/python/requirements-3.13.txt
+++ b/python/requirements-3.13.txt
@@ -1,0 +1,18 @@
+humanize==4.9.0
+junitparser==3.1.2
+lxml==5.1.0
+psutil==5.9.8
+PyGithub==2.2.0
+  Deprecated==1.2.14
+    wrapt==1.16.0
+  PyJWT==2.8.0
+  PyNaCl==1.5.0
+    cffi==1.17.0rc1
+      pycparser==2.22
+  requests==2.32.3
+    certifi==2024.7.4
+    charset-normalizer==3.3.2
+    idna==3.7
+    urllib3==2.2.2
+  typing_extensions==4.12.2
+  urllib3==2.2.2

--- a/python/requirements-3.13.txt
+++ b/python/requirements-3.13.txt
@@ -7,7 +7,7 @@ PyGithub==2.2.0
     wrapt==1.16.0
   PyJWT==2.8.0
   PyNaCl==1.5.0
-    cffi==1.17.0rc1
+    cffi==1.17.1
       pycparser==2.22
   requests==2.32.3
     certifi==2024.7.4

--- a/python/requirements-3.7.txt
+++ b/python/requirements-3.7.txt
@@ -3,7 +3,7 @@ humanize==4.6.0
     typing_extensions==4.7.1
     zipp==3.15.0
 junitparser==3.1.2
-lxml==5.1.0
+lxml==5.3.0
 psutil==5.9.8
 PyGithub==2.2.0
   Deprecated==1.2.14

--- a/python/requirements-3.8.txt
+++ b/python/requirements-3.8.txt
@@ -1,6 +1,6 @@
 humanize==4.9.0
 junitparser==3.1.2
-lxml==5.1.0
+lxml==5.3.0
 psutil==5.9.8
 PyGithub==2.2.0
   Deprecated==1.2.14

--- a/python/requirements-3.8.txt
+++ b/python/requirements-3.8.txt
@@ -7,7 +7,7 @@ PyGithub==2.2.0
     wrapt==1.16.0
   PyJWT==2.8.0
   PyNaCl==1.5.0
-    cffi==1.16.0
+    cffi==1.17.1
       pycparser==2.22
   requests==2.32.3
     certifi==2024.7.4

--- a/python/requirements-3.9.txt
+++ b/python/requirements-3.9.txt
@@ -1,6 +1,6 @@
 humanize==4.9.0
 junitparser==3.1.2
-lxml==5.1.0
+lxml==5.3.0
 psutil==5.9.8
 PyGithub==2.2.0
   Deprecated==1.2.14

--- a/python/requirements-3.9.txt
+++ b/python/requirements-3.9.txt
@@ -7,7 +7,7 @@ PyGithub==2.2.0
     wrapt==1.16.0
   PyJWT==2.8.0
   PyNaCl==1.5.0
-    cffi==1.16.0
+    cffi==1.17.1
       pycparser==2.22
   requests==2.32.3
     certifi==2024.7.4

--- a/python/requirements-post-3.7.txt
+++ b/python/requirements-post-3.7.txt
@@ -1,6 +1,6 @@
 humanize==4.9.0
 junitparser==3.1.2
-lxml==5.1.0
+lxml==5.3.0
 psutil==5.9.8
 PyGithub==2.2.0
   Deprecated==1.2.14

--- a/python/requirements-post-3.7.txt
+++ b/python/requirements-post-3.7.txt
@@ -7,7 +7,7 @@ PyGithub==2.2.0
     wrapt==1.16.0
   PyJWT==2.8.0
   PyNaCl==1.5.0
-    cffi==1.17.0rc1
+    cffi==1.17.1
       pycparser==2.22
   requests==2.32.3
     certifi==2024.7.4

--- a/python/requirements-post-3.7.txt
+++ b/python/requirements-post-3.7.txt
@@ -7,7 +7,7 @@ PyGithub==2.2.0
     wrapt==1.16.0
   PyJWT==2.8.0
   PyNaCl==1.5.0
-    cffi==1.16.0
+    cffi==1.17.0rc1
       pycparser==2.22
   requests==2.32.3
     certifi==2024.7.4

--- a/python/requirements.txt
+++ b/python/requirements.txt
@@ -1,6 +1,6 @@
 humanize==4.6.0; python_version <= '3.7'
 humanize==4.9.0; python_version >  '3.7'
 junitparser==3.1.2
-lxml==5.1.0
+lxml==5.3.0
 psutil==5.9.8
 PyGithub==2.2.0

--- a/windows/action.yml
+++ b/windows/action.yml
@@ -193,7 +193,7 @@ runs:
       continue-on-error: true
       with:
         path: '~\AppData\Local\pip\Cache'
-        key: enricomi-publish-action-${{ runner.os }}-${{ runner.arch }}-pip-${{ steps.python.outputs.version }}-e594996205319a7990b3a4ec677d10a3
+        key: enricomi-publish-action-${{ runner.os }}-${{ runner.arch }}-pip-${{ steps.python.outputs.version }}-1a7833c49f5935a970bc661e3c45aa2a
 
     - name: Create virtualenv
       id: venv

--- a/windows/bash/action.yml
+++ b/windows/bash/action.yml
@@ -194,7 +194,7 @@ runs:
       continue-on-error: true
       with:
         path: '~\AppData\Local\pip\Cache'
-        key: enricomi-publish-action-${{ runner.os }}-${{ runner.arch }}-pip-${{ steps.python.outputs.version }}-e594996205319a7990b3a4ec677d10a3
+        key: enricomi-publish-action-${{ runner.os }}-${{ runner.arch }}-pip-${{ steps.python.outputs.version }}-1a7833c49f5935a970bc661e3c45aa2a
 
     - name: Create virtualenv
       id: venv

